pkg/trace/writer/stats.go

```diff
@@ -140,9 +133,19 @@ func (w *StatsWriter) Stop() {
 	stopSenders(w.senders)
 }
 
+// Add appends this StatsPayload to the writer's buffer (flushing immediately if syncMode is enabled)
+func (w *StatsWriter) Add(sp *pb.StatsPayload) {
+	w.addStats(sp)
+	if !w.syncMode {
```

Comment:
huh, good point. It looks like the name here is confusing from the intended behavior. I found this comment on the incoming config `Mode where traces are only submitted when FlushAsync is called` it looks like this config exists so that serverless can have a "sync" mode where they can use Add (or previously the `In` channel) but they only want to flush the stats payloads when they call FlushSync explicitly and not just enqueueing them in the sender's queue. We could possible rename this to something like `ServerlessMode` or maybe `ExplicitSyncFlushOnly`... names are hard :(

Reply:
I've noticed this before. It does seem the name is backwards, but I think the logic is correct. It should just be renamed, I think.